### ISSUE TITLE
Preincrement and Predecrement

### ISSUE THREAD

**Issue by mkristien** [Hyp-ed/styleguide#2]
This seems unnecessarily. Do we ever use complex iterators? I would just stick with `i++` or let people use whatever.

**Comment by neilweidinger** [comment]
I agree too, using postincrement everywhere is probably the best way to go for the sake of consistency and readability, but if there's a decent reason to use preincrement that's fine as well.

**Comment by brano2** [comment]
What about the efficiency? `++i` is more efficient in for-loops. Or does the compiler optimise `i++` in that case so that it doesn't matter? Are we using an appropriate level of compiler optimisation for this to happen?
